.github/workflows/formatting.yaml

```diff
@@ -0,0 +1,34 @@
+name: Formatting
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+    types:
+      - opened
+      - synchronize
+
+jobs:
+  code-format-checks:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - uses: DoozyX/clang-format-lint-action@v0.14
+        with:
+          source: '.'
+          extensions: 'h,c,cc,cpp,proto,java'
+          clangFormatVersion: 14
+      - name: Set up Python
+        uses: actions/setup-python@v2
+        with:
+          python-version: 3.11
+      - name: Install dependencies
+        run: |
+          pip install 'black==24.10.0' 'isort==5.13.2'
```

Comment:
Any particular reason to choose black over ruff? Ruff can check formatting and isort in a single path with the correct config file.

Reply:
I'm old and tend to stick with what I know 😅

Reply:
Fair enough, lets go with this for now; I can upgrade to ruff at some point if I have spare cycles..